src/lib/chat/matrix-client.ts

```diff
@@ -104,6 +104,27 @@ export class MatrixClient implements IChatClient {
     return [];
   }
 
+  // NOTE: This can be removed in the next release cycle,
+  // since it will fix the "already existing" conversation groups,
+  // and we create a new room with the appropriate power_levels now.
+  private async setPowerLevels(rooms: Room[]) {
+    for (const room of rooms) {
+      const powerLevels = this.getLatestEvent(room, EventType.RoomPowerLevels);
+      if (!powerLevels || this.userId === room.getCreator()) {
+        continue;
+      }
+
+      const powerLevelsByUser = powerLevels.getContent()?.users || {};
+      for (const userId of Object.keys(powerLevelsByUser)) {
```

Comment:
Since we're only doing the current user can we just get their record rather than loop?
`powerLevelsByUser[this.userId]`?

Reply:
oh yes you're right. thanks for spotting!